runtime/norm/src/.component/.expand/version.js

```diff
@@ -1,19 +1,46 @@
 'use strict'
 
-function version (manifest) {
-  if (manifest.version === undefined) {
-    const bridge = require(manifest.bridge)
+const { join } = require('node:path')
+const { createHash } = require('node:crypto')
+const fs = require('node:fs/promises')
+const { createReadStream } = require('node:fs')
+const { once } = require('node:events')
 
-    if ('version' in bridge)
-      manifest.version = bridge.version(manifest)
-  }
+async function version (manifest) {
+  manifest.version ??= await hash(manifest.path)
+}
+
+async function hash (path) {
+  const hash = await hashd(path)
+  const hex = hash.digest('hex')
+
+  return hex.slice(0, 8)
+}
 
-  if (manifest.version === undefined) {
-    console.warn(`Component '${manifest.namespace ? manifest.namespace + '.' : ''}${manifest.name}' has no version`)
+/**
+ * @param {string} path
+ * @param {import('node:crypto').Hash} hash
+ */
+async function hashd (path, hash = createHash('sha256')) {
+  const stat = await fs.stat(path)
 
-    manifest.version = Math.random().toString(36).slice(2)
+  if (stat.isFile()) {
+    const stream = createReadStream(path)
+
+    stream.pipe(hash, { end: false })
+
+    return await once(stream, 'end')
   }
 
+  if (stat.isDirectory()) {
+    const entries = await fs.readdir(path)
+
+    for (const entry of entries) {
```

Comment:
NIT: [fs.opendir](https://nodejs.org/api/fs.html#fspromisesopendirpath-options) will be more memory efficient here as we are doing stream way